Could not get this adapter working via the example, without an existing policy to load

I am using this adapter to essentially have an in-memory policy. I was unable to continue without pre-seeding the byte buffer with `[]`:

```go
func (a *Adapter) loadFromBuffer() error {
	var policy []CasbinRule
	err := json.Unmarshal(*a.source, &policy)
	if err == nil {
		a.policy = policy
	}
	return err
}
```
err is `unexpected end of JSON input` which is expected given the code.

```go
func TestCasbinEnforcerWithInMemoryStoreAndNoSeededPolicy(t *testing.T) {
	modelStr := `
[request_definition]
r = sub, obj, act

[policy_definition]
p = sub, obj, act

[policy_effect]
e = some(where (p.eft == allow))

[matchers]
m = r.sub == p.sub && keyMatch(r.obj, p.obj) && r.act == p.act
`
	model, err := casbinModel.NewModelFromString(modelStr)
	if err != nil {
		t.Fatalf("failed to load model: %v", err)
	}

	var buf []byte
	adapter := jsonadapter.NewAdapter(&buf)

	enforcer, err := casbin.NewEnforcer(model, adapter)
	if err != nil {
		t.Fatalf("failed to initialize enforcer: %v", err)
	}

	ok, err := enforcer.AddPolicy("some_user", "some_scope", "some_permission")
	if err != nil {
		t.Fatalf("failed to add policy: %v", err)
	}

	if !ok {
		t.Error("policy was not added?")
	}
}
```

results with

```
=== RUN   TestCasbinEnforcerWithInMemoryStoreAndNoSeededPolicy
    casbin_test.go:43: failed to initialize enforcer: unexpected end of JSON input
--- FAIL: TestCasbinEnforcerWithInMemoryStoreAndNoSeededPolicy (0.00s)
```

--- a/adapter.go
+++ b/adapter.go
@@ -56,10 +56,11 @@ func (a *Adapter) saveToBuffer() error {
 }
 
 func (a *Adapter) loadFromBuffer() error {
-	var policy []CasbinRule
-	if len(*a.source) == 0{
+	if len(*a.source) == 0 {
 		return nil
 	}
+
+	var policy []CasbinRule
 	err := json.Unmarshal(*a.source, &policy)
 	if err == nil {
 		a.policy = policy

--- a/adapter.go
+++ b/adapter.go
@@ -57,6 +57,9 @@ func (a *Adapter) saveToBuffer() error {
 
 func (a *Adapter) loadFromBuffer() error {
 	var policy []CasbinRule
+	if len(*a.source) == 0{
+		return nil
+	}
 	err := json.Unmarshal(*a.source, &policy)
 	if err == nil {
 		a.policy = policy

--- a/go.mod
+++ b/go.mod
@@ -2,4 +2,4 @@ module github.com/casbin/json-adapter/v2
 
 go 1.12
 
-require github.com/casbin/casbin/v2 v2.0.0
+require github.com/casbin/casbin/v2 v2.60.0

